### PR TITLE
CLI: Add non-interactive option to interactive commands

### PR DESCRIPTION
**Pull request by GlancingMind**
Using an empty string for the avatar cli flag e.g. `git-bug user create -a ""` will still result in a prompt. As the avatar URL is an optional option, it should be possible to skip asking for it entirely.
Otherwise automated user creation via a script must make use of pipe hacks.

### CODE REVIEW

**Comment by MichaelMure** [comment]
That's indeed something I didn't think about. Definitely need to be addressed but after thinking about it for a while, maybe we need a more generic solution that we could apply to all commands?

Maybe something like `--no-interactive` that would disable all the prompts?

**Comment by GlancingMind** [comment]
Sounds like a better solution. I'll take a look.

**Comment by GlancingMind** [comment]
Found some inconsistency. The `user create` command requires email to be set (uses `input.required`), but `Validate()` in `identity/version.go` doesn't check for an empty email. This means, that currently a user can be created (non-interactively) without specifying an email. So I'm not sure, where I should check for the missing email or if I check at all.

**Comment by GlancingMind** [comment]
I've added a global option `non-interactive` which can be used in every git-bug (sub-)command.
The `user create` command won't fail if `--non-interactive` is set and no email is given.

**Comment by GlancingMind** [comment]
Went through all commands and checked for those with interactive prompts, the interactive mode.
Except the bridges itself. Don't know, if I should alter them or the respective bridge author/commiter.
Otherwise, I hope that I didn't miss one.

**Comment by MichaelMure** [comment]
> Don't know, if I should alter them or the respective bridge author/commiter.

Yes please!

File: commands/root.go [hunk 1]
```diff
@@ -15,6 +15,12 @@ var GitCommit string
 var GitLastTag string
 var GitExactTag string
 
+type GlobalOptions struct {
+	nonInteractive bool
+}
+
+var rootOptions GlobalOptions
```
Comment (MichaelMure) [inline commands/root.go:22]: I get why you did  that but we really should avoid global variables like that. At the beginning of git-bug each flags were done as global variables like that (following the cobra standard) but it ended up being a major pain. Following that came this new "functional" structure without any globals.

Now to get out of this, there still might be a way to define the flag at the root level (say, with `cmd.PersistentFlags().Bool("non-interactive", ....)`, but it looks to me that we could just avoid that entirely and define the flag only on the commands where it makes sense (that is, those that have prompts).

What do you think?

**Review by MichaelMure — CHANGES_REQUESTED** [review]
As explained in  the comment, we should avoid the global variable.

File: commands/root.go (hunk 1 above)
Comment (GlancingMind) [reply commands/root.go:22]: I have the same feeling about the current solution. Good that we are on the same wavelength.
I will remove the global option and instead add a dedicated non-interactive option on each of the interactive commands.
Shouldn't be to much work.

**Comment by GlancingMind** [comment]
> > Don't know, if I should alter them or the respective bridge author/commiter.
> 
> Yes please!

I can think about two ways to implement this.
1. Pass the non-interactive flag to Bridge::Configure(). Then the bridge has to check for the flag itself, before making a prompt.
2. Make the input.go/prompt.go an interface. Then create a *non-interactive*-implementation which basically prints an error instead of prompting for user input. The `bridge configure` command will then pass this *non-interactive* input implementation to Bridge::Configure(). Now the specific bridges won't have to deal with the non-interactive option, as long as they are using the git-bug/input package. (Not sure how to implement this in go, have to read myself into it.)

I think the 2. option is the better one, but wanted your (@MichaelMure) feedback first.

**Comment by MichaelMure** [comment]
I would go for (1), as an extra boolean flag to `Configure`. Let's keep the prompts being prompts instead of adding a new feature there.

**Comment by GlancingMind** [comment]
- [x] GitHub bridge
- [ ] GitLab bridge
- [ ] Jira
- [ ] launchpad

**Comment by GlancingMind** [comment]
So fare I'm done. Hope that I got everything right.

@MichaelMure are you able to test the JIRA bridge authentification?

**Comment by MichaelMure** [comment]
I changed the logic in the bridges from `isNonInteractive` --> `interactive`, good to go now! Thanks!

**Pull request merged**